recipes/yojimbo/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, CMake, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class YojimboConan(ConanFile):
+    name = "yojimbo"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/networkprotocol/yojimbo"
+    topics = ("conan", "yojimbo", "game", "udp", "protocol", "client-server", "multiplayer-game-server")
+    description = "A network library for client/server games written in C++"
+    license = "BSD-3-Clause"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    build_requires = "premake/5.0.0-alpha15"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libsodium/1.0.18")
+        self.requires("mbedtls/2.25.0")
+ 
+    def source(self):
+        git = tools.Git(folder=self._source_subfolder)
+        git.clone(self.homepage, branch="v"+self.version, shallow=True)
+        git.run("submodule update --init --recursive --recommend-shallow")
```

Comment:
We typically do not work this way on CCI, you can check the docs https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#the-conandatayml

It's in part why the build is failing.

You'll need to add the two subprojects as their own recipes first, that's been our rule thus far

Reply:
The issue with adding the files as sub-projects is that the sub-module dependencies are not built separately and linked to the library, the source is directly included. Is there a way to deal with this in the conan recipe without changing the project's premake file significantly?

Reply:
Yeah this project is very different than most OSS so it's not straightforward to bring this into CCI (as a community reviewer) You might want to try the slack channel and see if you can get more ideas there.

I see the problem https://github.com/networkprotocol/yojimbo/blob/48197c484ba7b8666459f45418fd8708e8a79fac/premake5.lua#L36 directly includes the code from the submodules. 

The best example I can give you is #3786 which uses a https://github.com/conan-io/conan-center-index/blob/master/recipes/open62541/all/submoduledata.yml to download the correct versions. As I recall the CMake build script downloads the submodules.

Since the submodules match the releases of the sub-projects you could define multiple sources in the `conandata.yml` (think this is the closest we can get to normal).

1. download the release for `yojimbo` 
2. download each submodule to the correct folder.

#5297 is an example of downloading multiple items from the `conandata.yml`

***

I also found the an example using premake, https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/nativefiledialog/all/conanfile.py (by the Conan team so it's the best reference).

There's a few extra CMake references in your recipe which can be removed.